File: tks/RepositoriesAdapters/src/main/kotlin/mappers/UserMapper.java
package mappers;

import data.UserEntity;
import model.User;
import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;
import org.mapstruct.factory.Mappers;

@Mapper
public interface UserMapper {

    UserMapper INSTANCE = Mappers.getMapper(UserMapper.class);

    User mapEntityToDomainObject(UserEntity entity);

    UserEntity mapDomainObjectToEntity(User user);

    void mapDomainObjectToEntity(User user, @MappingTarget UserEntity entity);
}

File: tks/RepositoriesAdapters/src/main/kotlin/mappers/UserMapper.kt
package mappers

import common.mappers.EmailMapper
import data.UserEntity
import org.mapstruct.MappingTarget
import domain.model.User
import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers

@Mapper(uses =[EmailMapper::class])
interface UserMapper {
    fun mapEntityToDomainObject(entity: UserEntity?): User?
    fun mapDomainObjectToEntity(user: User?): UserEntity?
    fun mapDomainObjectToEntity(user: User?, @MappingTarget entity: UserEntity?)

    companion object {
        val INSTANCE:UserMapper = Mappers.getMapper(UserMapper::class.java)
    }
}